### PR TITLE
make sure tmpdir resolvs user dirs

### LINKED ISSUE #20332: file transfer failure after upgrade from 2.2.0 to 2.2.1
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Local connection plugin?

##### ANSIBLE VERSION

```
ansible 2.2.1.0
  config file =
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

```
[ssh_connection]
# http://docs.ansible.com/intro_configuration.html#pipelining
pipelining=True
# http://docs.ansible.com/intro_configuration.html#ssh-args
ssh_args = -o ControlMaster=auto -o ControlPersist=30m
forks = 50
```

##### OS / ENVIRONMENT

`16.04.1 LTS (Xenial Xerus)`

##### SUMMARY

When `cloud-init` runs `ansible-playbook -i hosts -c local /tmp/ansible/playbooks/env_production.yml --limit=bastion`, I get the following error:
```
PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
fatal: [bastion]: FAILED! => {"failed": true, "msg": "failed to transfer file to ~/.ansible/tmp/ansible-tmp-1484618416.23-141753847363017/setup.py: [Errno 2] No such file or directory: '~/.ansible/tmp/ansible-tmp-1484618416.23-141753847363017/setup.py'"}
        to retry, use: --limit @/tmp/ansible/playbooks/env_production.retry

PLAY RECAP *********************************************************************
bastion : ok=0    changed=0    unreachable=0    failed=1
```

This doesn't happen Ansible 2.2.0. No other changes have been done.
This doesn't happen if I run the script manually from a shell.

The only way I have to reproduce the error is to terminate and spawn a new EC2 instance.
I'm trying to find another way to reproduce it.

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

shell

##### ANSIBLE VERSION

```
2.3
```

##### SUMMARY

fixes #20332
supercedes #20484

### CODE REVIEW

File: lib/ansible/plugins/action/__init__.py [hunk 1]
```diff
@@ -218,7 +218,12 @@ def _make_tmp_path(self, remote_user):
 
         tmp_mode = 0o700
 
-        cmd = self._connection._shell.mkdtemp(basefile, use_system_tmp, tmp_mode)
+        if use_systemp_tmp:
```
Comment (MiLk) [inline lib/ansible/plugins/action/__init__.py]: Typo here: `use_system_tmp`

**Review by MiLk — CHANGES_REQUESTED** [review]
Except the typo, it's working.
Good job.

**Review by MiLk — APPROVED** [review]

**Comment by MiLk** [comment]
:shipit:

**Comment by MiLk** [comment]
Can we get this merged, backported to 2.2 and expect a new 2.2 release soon?

**Comment by bcoca** [comment]
probably, yes, maybe

**Comment by abadger** [comment]
+1

**Pull request merged**